lib/src/log_handler.dart

```diff
@@ -0,0 +1,99 @@
+import 'dart:convert';
+
+import 'package:file/file.dart';
+import 'package:path/path.dart' as p;
+
+import 'constants.dart';
+import 'initializer.dart';
+
+class LogFileStats {
+  /// The oldest timestamp in the log file
+  final DateTime startDateTime;
+
+  /// The latest timestamp in the log file
+  final DateTime endDateTime;
+
+  /// The number of unique session ids found in the log file
+  final int sessionCount;
+
+  /// Contains the data from the [LogHandler.logFileStats]
+  LogFileStats({
+    required this.startDateTime,
+    required this.endDateTime,
+    required this.sessionCount,
+  });
+
+  @override
+  String toString() => jsonEncode(<String, dynamic>{
+        'startDateTime': startDateTime.toString(),
+        'endDateTime': endDateTime.toString(),
+        'sessionCount': sessionCount,
+      });
+}
+
+class LogHandler {
+  final FileSystem fs;
+  final Directory homeDirectory;
+  final File logFile;
+
+  /// This class is responsible for writing to a log
+  /// file that has been initialized by the [Initializer]
+  ///
+  /// It will be treated as an append only log and will be limited
+  /// to have has many data records as specified by [kLogFileLength]
+  LogHandler({
```

Comment:
Although, does this really need to be a class? Or can its methods just be methods of the `Analytics` class?

Reply:
I prefer to have all the logic for the files this package interacts with in a class to make it easier to understand what is happening. I have done this with the config file with `ConfigHandler` and the session.json file also has `Session` to handle any session related operations

Reply:
> I prefer to have all the logic for the files this package interacts with in a class to make it easier to understand what is happening. I have done this with the config file with `ConfigHandler` and the session.json file also has `Session` to handle any session related operations

I think it's ultimately down to preference, so it's fine as is. I'm just pointing out the down-side to having a tree structure (where the `Analytics` class is the root node, and it has several dependent objects such as `ConfigHandler` and `LogHandler`) is that the end to end logic of (for example) sendEvent is less obvious, and you've created more boiler plate for yourself in terms of constructing all the objects.

> Also I am unable to make this a `const` constructor because the `logFile` class variable is not a constant value

Good point, I missed that.

Reply:
That is a good point, we can look into possibly simplifying things in the future if we find that the log handler is only responsible for the save and logfilestats methods, but it could be a good idea now if we need add additional functionality as we onboard dash tools